Source/Dafny/Compilers/Compiler-Csharp.cs

```diff
@@ -123,6 +123,10 @@ public class CsharpCompiler : SinglePassCompiler {
     }
 
     protected override void EmitBuiltInDecls(BuiltIns builtIns, ConcreteSyntaxTree wr) {
+      if (builtIns.MaxNonGhostTupleSizeUsed > 20) {
+        Reporter!.Error(MessageSource.Compiler, builtIns.MaxNonGhostTupleSizeToken, "C# back-end does not support tuples with more than 20 arguments.");
```

Comment:
Any reason this doesn't just call `Error(...)`? https://github.com/dafny-lang/dafny/blob/master/Source/Dafny/Compilers/SinglePassCompiler.cs#L76

Only nit-picking here because it will make adapting this to the changes I have pending for https://github.com/dafny-lang/dafny/issues/1770 easier.

Reply:
I saw that but it didn't seem to do anything useful unless you have a ConcreteSyntaxTree to write to. Maybe you can make Reporter inaccessible in your PR to prevent accidentally missing the logic you're adding?

Reply:
Oh I can easily handle it if it doesn't make sense to call it here.

Now that I look closer, `Error(...)` is semi-broken because it assumes `/* ... */` is valid comment syntax in the target language. I'm not convinced it's worth the trouble to output errors in the target program in addition to reporting them, but I'll deal with that in my own PR.

Reply:
> Now that I look closer, Error(...) is semi-broken because it assumes /* ... */ is valid comment syntax

https://github.com/dafny-lang/dafny/issues/1729

Reply:
Nice, thanks! Assigned it to myself, will do something about it as I finish the work on #1770